integration-tests/dns/lib.rs

```diff
@@ -115,13 +118,13 @@ mod dns {
         #[ink(message)]
         pub fn set_address(&mut self, name: Hash, new_address: AccountId) -> Result<()> {
             let caller = self.env().caller();
-            let owner = self.get_owner_or_default(name);
+            let owner = self.get_owner_or_default(name)?;
             if caller != owner {
                 return Err(Error::CallerIsNotOwner)
             }
 
             let old_address = self.name_to_address.get(name);
-            self.name_to_address.insert(name, &new_address);
+            self.name_to_address.try_insert(name, &new_address).map_err(|_| Error::ValueTooLarge)?;
```

Comment:
I was also thinking if it makes sense.
I think you are right; I will implement it only for the 'multisig' contract, where transactions (scale-encoded data) are kept in a Mapping.

Reply:
Yes the use case for the fallible API is rather narrow, in a majority of cases it shouldn't be necessary at all. 
However it is covered in the [mapping integration test](https://github.com/paritytech/ink/blob/master/integration-tests/mapping-integration-tests/lib.rs#L113)?

Reply:
For `Mapping` and `StorageVec`, it is recommended to use the fallible API whenever the data _can_ overgrow the static buffer size and that case needs to be handled.

In practice this means it's relevant when dynamically sized types (or to be pedantic: types with non static encoding size) are used.

However just because some type is of dynamic size, the fallible API is not automatically necessary, because in practice the type can't overgrow the static buffer despite it being dynamic.

Examples:
- A mapping `M` of aliases to balances, where aliases are strings. Given at all relevant places, the contract makes sure that (the encoding size of) aliases are never longer than some maximum `MSize`. Given `BufferSize` >= `MSize` ++ `Balance`, operations on mapping `M` can never panic 
- The contract checks that the data fits before calling the storage opration.
- The is known to fit into the static buffer, for example because it came as a function argument, it wasn't modified and the total encoded size of the message selector plus the all function arguments do not exceed the static buffer size. (The data _was already_ in the static buffer and is not modified. So it's guaranteed to fit again)

See also the [(unreleased) section in the ink docs](https://use.ink/5.x/datastructures/mapping#use-fallible-storage-methods-for-dynamically-sized-values)